server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2352,6 +2352,17 @@ public Translog.Snapshot getHistoryOperations(String reason, long startingSeqNo,
         return getEngine().newChangesSnapshot(reason, startingSeqNo, endSeqNo, true, accurateCount);
     }
 
+    /**
+     * Creates a new history snapshot from the translog instead of the lucene index. Required for cross cluster replication.
+     * Use the recommended {@link #getHistoryOperations(String, long, long, boolean)} method for other cases.
+     * Depending on how translog durability is configured, this method might/might not return the snapshot. For e.g,
+     * If the translog has been configured with no-durability, it'll return UnsupportedOperationException.
+     */
+    public Translog.Snapshot getHistoryOperationsFromTranslog(long startingSeqNo, long endSeqNo) throws IOException,
```

Comment:
Since this is only for CCR we should mark this as `@deprecated`.

Reply:
Instead, I've added an assertion on this method to ensure that we don't call this for Indices with SegRep or remote store enabled.